db/migrations/2024_04_20_000000_refactor_permissions_and_groups.php

```diff
@@ -0,0 +1,205 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Engelsystem\Migrations;
+
+use Engelsystem\Database\Migration\Migration;
+use Illuminate\Database\Connection;
+use Illuminate\Database\Schema\Builder as SchemaBuilder;
+
+class RefactorPermissionsAndGroups extends Migration
+{
+    protected int $developer = 90;
+    protected int $bureaucrat = 80;
+    protected int $shiCo = 60;
+    protected int $newsAdmin = 85;
+    protected int $teamCoordinator = 65;
+    protected int $angel = 20;
+
+    protected int $active;
+    protected int $driveEdit;
+    protected int $eventConfig;
+    protected int $goodieEdit;
+    protected int $ifsgEdit;
+    protected int $log;
+    protected int $news;
+    protected int $register;
+    protected int $scheduleImport;
+    protected int $shifts;
+    protected int $user;
+    protected int $userAngeltypes;
+    protected int $userShifts;
+
+    protected string $shiftentry = 'shiftentry_edit_angeltype_supporter';
+    protected string $language = 'admin_language';
+    protected string $userEdit = 'user.edit';
+    protected string $userNickEdit = 'user.nick.edit';
+    protected string $shifttypes = 'shifttypes';
+    protected string $shifttypesView = 'shifttypes.view';
+
+    protected Connection $db;
+
+    public function __construct(SchemaBuilder $schema)
+    {
+        parent::__construct($schema);
+        $this->db = $this->schema->getConnection();
+
+        $this->active = $this->getPrivilegeId('admin_active');
```

Comment:
Having a mapping of `'permission.name' => 'id'` would make the code below imho more readable but thats an option too